test/models/ht_user_test.rb

```diff
@@ -3,13 +3,16 @@
 require 'test_helper'
 
 class HTUserTest < ActiveSupport::TestCase
-  test 'iprestrict validation passes' do
-    user = HTUser.new(userid: 'user', iprestrict: '127.0.0.1')
+  valid_expires = '2020-09-30 16:03:09'
+  valid_iprestrict = '127.0.0.1'
+
+  test 'validation passes' do
+    user = HTUser.new(userid: 'user', iprestrict: valid_iprestrict, expires: valid_expires)
     assert user.valid?
   end
 
   test 'iprestrict validation fails' do
-    user = HTUser.new(userid: 'user', iprestrict: '127.0.0.1.1')
+    user = HTUser.new(userid: 'user', iprestrict: '127.0.0.1.1', expires: valid_expires)
```

Comment:
Having to add parameters to the constructor each time you add a required field seems like a pain. I would implement a factory method in your helpers (either yourself or using something like FactoryBot or Fabricator) that has `valid_expires` and `valid_iprestrict` as default parameters but lets you override them. Let me know if you need a concrete example of that.

Reply:
Rebased against your comments and force-pushed.